common/src/main/java/com/microsoft/identity/common/internal/controllers/LocalMSALController.java

```diff
@@ -400,6 +393,42 @@ public AcquireTokenResult acquireTokenSilent(
         return acquireTokenSilentResult;
     }
 
+    private void setAcquireTokenResult(final AcquireTokenResult acquireTokenSilentResult,
+                                       final SilentTokenCommandParameters parametersWithScopes,
+                                       final List<ICacheRecord> cacheRecords) throws ClientException {
+        ICacheRecord fullCacheRecord = cacheRecords.get(0);
+        acquireTokenSilentResult.setLocalAuthenticationResult(
+                new LocalAuthenticationResult(
+                        finalizeCacheRecordForResult(
+                                fullCacheRecord,
+                                parametersWithScopes.getAuthenticationScheme()
+                        ),
+                        cacheRecords,
+                        SdkType.MSAL,
+                        true
+                )
+        );
+    }
+
+    private void renewAT(@NonNull final SilentTokenCommandParameters parametersWithScopes,
+                         @NonNull final AcquireTokenResult acquireTokenSilentResult,
+                         @SuppressWarnings(WarningType.rawtype_warning) @NonNull final OAuth2TokenCache tokenCache,
+                         @SuppressWarnings(WarningType.rawtype_warning) @NonNull final OAuth2Strategy strategy,
+                         @NonNull final ICacheRecord cacheRecord,
+                         @NonNull final String tag) throws IOException, ClientException, ServiceException {
+        Logger.verbose(
+                TAG + tag,
+                "Renewing access token..."
+        );
+        renewAccessToken(
+                parametersWithScopes,
+                acquireTokenSilentResult,
+                tokenCache,
+                strategy,
+                cacheRecord
+        );
+    }
```

Comment:
I don't see anything special happening here....why do we need this method?

Reply:
Duplicate code.

Reply:
You mean to avoid duplicate code or to intentionally duplicate code? I am personally not a fan of adding a new method for sole purpose of injection a tag (I believe that is all you're trying to do here)

Reply:
If you look this method is used twice within the Controller so the method is reducing the amount of code.